thewalrus/reference.py

```diff
@@ -278,3 +282,121 @@ def hafnian(M, loop=False):
         tot_sum = tot_sum + result
 
     return tot_sum
+
+
+def mapper(x, objects):
+    """Helper function to turn a permutation and bistring into an element of rpmp.
```

Comment:
It'd be nice to add a little example here.

Reply:
Like so:
```python
mapper(((1,2,3), '0000'), (0,1,2,3,4,5,6,7))
Out[38]: ((0, 5), (1, 6), (2, 7), (4, 3))
```
?

Reply:
yes!

Reply:
I find this function particularly hard to follow. It would be nice to have a test that specifically checks the output of this function against hard-coded values.

Reply:
Added an extra test `test_mapper_hard_coded`.